protoc-gen-openapiv2/internal/genopenapi/template.go

```diff
@@ -1004,14 +1015,72 @@ pathLoop:
 			}
 			// Pop from the stack
 			depth--
-			buffer += string(char)
-			if reg.GetUseJSONNamesForFields() &&
-				len(jsonBuffer) > 1 {
-				jsonSnakeCaseName := jsonBuffer[1:]
-				jsonCamelCaseName := lowerCamelCase(jsonSnakeCaseName, fields, msgs)
-				prev := buffer[:len(buffer)-len(jsonSnakeCaseName)-2]
-				buffer = strings.Join([]string{prev, "{", jsonCamelCaseName, "}"}, "")
-				jsonBuffer = ""
+			paramName := ""
+			pattern := ""
+			if reg.GetExpandSlashedPathPatterns() {
+				paramPattern := strings.SplitN(buffer[1:], "=", 2)
+				if len(paramPattern) == 2 {
+					paramName = paramPattern[0]
+					pattern = paramPattern[1]
+				}
+			}
+			if pattern == "" || pattern == "*" {
+				buffer += string(char)
+				if reg.GetUseJSONNamesForFields() &&
+					len(jsonBuffer) > 1 {
+					jsonSnakeCaseName := jsonBuffer[1:]
+					jsonCamelCaseName := lowerCamelCase(jsonSnakeCaseName, fields, msgs)
+					prev := buffer[:len(buffer)-len(jsonSnakeCaseName)-2]
+					buffer = strings.Join([]string{prev, "{", jsonCamelCaseName, "}"}, "")
+					jsonBuffer = ""
+				}
+				continue
+			}
+
+			pathParamIndex := slices.IndexFunc(*pathParams, func(p descriptor.Parameter) bool {
+				return p.FieldPath.String() == paramName
+			})
+			if pathParamIndex == -1 {
+				panic(fmt.Sprintf("Path parameter %q not found in path parameters", paramName))
+			}
+			pathParam := (*pathParams)[pathParamIndex]
+			patternParts := strings.Split(pattern, "/")
+			for i, part := range patternParts {
+				modifiedPart := part
+				if part == "*" {
+					paramName := strings.TrimSuffix(parts[len(parts)-1], "s")
```

Comment:
The wildcards from the path pattern create new path parameters, I decided to infer a name for these parameters from the previous path segment, as this works well for _my_ current use case, but I understand that this problematic for many reasons so I would like to discuss how to name the new path params?

Reply:
Yeah, obviously that heuristic isn't always going to work. Not that I have a better suggestion to be honest. I think most users of this pattern will be using the AIP guidelines, so this should work for them. Lets just make sure we document this clearly.